### DISCUSSION TITLE
How to update electron-builder version with cordova-electron: 3.0.1

### DISCUSSION THREAD

**Discussion by BenLaKnet** [apache/cordova#462] [Q&A]
Hi,

I obtained issues with cordova-electron.

I have this error message when I tried : cordova build electron --release

```
 • electron-builder  version=22.14.13 os=10.0.19045
  • loaded configuration  file=package.json ("build" field)
  • writing effective config  file=platforms\electron\build\builder-effective-config.yaml
TypeError: The "code" argument must be of type number. Received type string ('ERR_ELECTRON_BUILDER_INVA...')
    at process.set [as exitCode] (node:internal/bootstrap/node:123:9)
    at c:\Users\vero\AppData\Roaming\npm\node_modules\cordova\bin\cordova:32:22
```

I installed electron-builder: 24.9.1

But when I tried to build cordova always used version=22.14.13

How to change that ? or correct the error message.

Thanks.

**Comment by pjoriginal** [comment]
Is there a reason you'd like to update your `electron-builder` to a newer version?
If it's for compatibility with plugins and packages, I use [electron-rebuild](https://github.com/electron/rebuild) to rebuild packages with different versions so that they are in line with the electron version I'm using.

Edit: I re-read your question and see that you're stuck at the build part for bootstrap. the rebuild solution might work for you here. give it a try

**Comment by BenLaKnet** [comment]
Have you an example of using CLI ?

The problem is with cordova and electron config.

**Comment by pjoriginal** [comment]
@BenLaKnet 
Sorry I forgot to reply to this.
`cordova-electron` uses the `electron-builder` in its `package.json` file..
I've added steps here that would make it possible for you to change the electron-builder version to the one you want
```
cordova create test
cordova platform add electron@3.1.0
cd ./node_modules/cordova-electron
npm pkg set dependencies.electron-builder="^24.9.1"
cd ../..
cordova build electron --release
```

You can turn that into a script file and and attach it to the `before_prepare` or `after_platform_add` and you'd have a temporary build set up that builds using the builder version you want.

The new `cordova-electron` version is slated to release soon which has a picker for the electron version and the builder version
The PR is here https://github.com/apache/cordova-electron/pull/227

I think you should be able to use nightly if you don't depend on any plugins that require cordova-electron@3.x
and do:
```
"overrides": {
  "cordova-electron": {
    "electron-builder": "^23.0.3"
  }
}
```

**Reply by erisu** [reply]
To follow up on the previous solutions, consider the following alternative:

```bash
cordova create test
cd test
npm pkg set overrides.cordova-electron.electron-builder="^24.9.1"
cordova platform add electron@latest
cordova build electron --release
```

Before installing the `cordova-electron` platform, update the `package.json` file to include the `overrides` setting using the `npm pkg set` CLI command:

```bash
npm pkg set overrides.cordova-electron.electron-builder="^24.9.1"
```

This command will insert the following block into the `package.json` file:

```json
"overrides": {
  "cordova-electron": {
    "electron-builder": "^24.9.1"
  }
}
```

This approach allows npm CLI to handle the file update instead of manual editing.

Once the `overrides` are set, when adding the platform, it will fetch the version defined in the `overrides` property. Since `package.json` is versioned, if you are using a CI service to build, it will also know to fetch the overridden version.

Ensure you are using npm 8 or greater to support the `overrides` functionality.

**Comment by BenLaKnet** [comment]
Hi, many thanks for your answers. I tried and it returns an error:

```
  • electron-builder  version=24.12.0 os=10.0.19045
  • loaded configuration  file=package.json ("build" field)
  • writing effective config  file=platforms\electron\build\builder-effective-config.yaml
TypeError: The "code" argument must be of type number. Received type string ('ERR_ELECTRON_BUILDER_INVA...')
    at process.set [as exitCode] (node:internal/bootstrap/node:123:9)
    at c:\Users\vero\AppData\Roaming\npm\node_modules\cordova\bin\cordova:32:22
```
